services/cognito/spec/requests/users_spec.rb

```diff
@@ -31,6 +31,38 @@
         expect_json_sizes('data', models_count)
       end
     end
+
+    context 'perform query' do
+      include_context 'authorized user'
+
+      let(:models_count) { rand(1..5) }
+      let!(:models) { create_list :user, models_count }
+
+      context 'based on ID' do
+        before do
+          get url + "?filter[query]=#{models.last.id}", headers: request_headers
+        end
+
+        it 'returns and ok response with the user ID queried' do
+          expect(response).to have_http_status(:ok)
+          expect_json_sizes('data', 1)
```

Comment:
Here we should ensure that the returned result is actually matching the id.
You can do something like:
`expect_json('data.0', id: model.last.id.to_s)`